README.md

```diff
@@ -1,3 +1,139 @@
+    // at least one positional argument must be passed.
+    #[positional(1..)]
+    Text(String),
+}
+
+#[derive(Initial)]
+struct Settings {
+    // We can change the default value with the field attribute.
+    #[initial(true)]
+    caps: bool,
+    exclamation_marks: u8,
+    text: String,
+}
+
+// To implement `Options`, we only need to provide the `apply` method.
+// The `parse` method will be automatically generated.
+impl Options<Arg> for Settings {
+    fn apply(&mut self, arg: Arg) {
+        match arg {
+            Arg::NoCaps => self.caps = false,
+            Arg::ExclamationMarks(n) => self.exclamation_marks += n,
+            Arg::Text(s) => {
+                if self.text.is_empty() {
+                    self.text.push_str(&s);
+                } else {
+                    self.text.push(' ');
+                    self.text.push_str(&s);
+                }
+            }
+        }
+    }
+}
+
+fn run(args: &'static [&'static str]) -> String {
+    let s = Settings::parse(args);
+    let mut output = if s.caps {
+        s.text.to_uppercase()
+    } else {
+        s.text
+    };
+    for i in 0..s.exclamation_marks {
+        output.push('!');
+    }
+    output
+}
+
+// The first argument is the binary name. In this example it's ignored.
+assert_eq!(run(&["shout", "hello"]), "HELLO");
+assert_eq!(run(&["shout", "-e3", "hello"]), "HELLO!!!");
+assert_eq!(run(&["shout", "-e", "3", "hello"]), "HELLO!!!");
+assert_eq!(run(&["shout", "--no-caps", "hello"]), "hello");
+assert_eq!(run(&["shout", "-e3", "-n", "hello"]), "hello!!!");
+assert_eq!(run(&["shout", "-e3", "hello", "world"]), "HELLO WORLD!!!");
+```
+
+## Additional functionality
+
+To make it easier to implement [`Arguments`] and [`Options`], there are
+two additional traits:
+
+- [`Initial`] is an alternative to the [`Default`] trait from the standard
+  library, with a richer derive macro.
+- [`Value`] allows for easy parsing from `OsStr` to any type
+  implementing [`Value`]. This crate also provides a derive macro for
+  this trait.
+
+## Examples
+
+The following files contain examples of commands defined with
+`uutils_args`:
+
+- [hello world](https://github.com/tertsdiepraam/uutils-args/blob/main/examples/hello_world.rs)
+- [arch](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/arch.rs)
+- [b2sum](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/b2sum.rs)
+- [base32](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/base32.rs)
+- [basename](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/basename.rs)
+- [cat](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/cat.rs)
+- [echo](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/echo.rs)
+- [ls](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/ls.rs)
+- [mktemp](https://github.com/tertsdiepraam/uutils-args/blob/main/tests/coreutils/mktemp.rs)
```

Comment:
```suggestion
- [hello world](/uutils/uutils-args/blob/main/examples/hello_world.rs)
- [arch](/uutils/uutils-args/blob/main/tests/coreutils/arch.rs)
- [b2sum](/uutils/uutils-args/blob/main/tests/coreutils/b2sum.rs)
- [base32](/uutils/uutils-args/blob/main/tests/coreutils/base32.rs)
- [basename](/uutils/uutils-args/blob/main/tests/coreutils/basename.rs)
- [cat](/uutils/uutils-args/blob/main/tests/coreutils/cat.rs)
- [echo](/uutils/uutils-args/blob/main/tests/coreutils/echo.rs)
- [ls](/uutils/uutils-args/blob/main/tests/coreutils/ls.rs)
- [mktemp](/uutils/uutils-args/blob/main/tests/coreutils/mktemp.rs)
```

Reply:
The links were absolute because they should also work on docs.rs, not just on GitHub. I've fixed all occurrences in a new commit.